### DISCUSSION TITLE
worker to run Laravel console command instead of supervisor

### DISCUSSION THREAD

**Discussion by marayshi** [brefphp/bref#846] [Support]
Hi

I'm trying to deploy my laravel project with bref
every things working well like Queue worker.php
I know worker.php replace the need for supervisor to execute listener "php artisan queue:work"
I'm trying to so similar function with worker for different console command to run 24/7 with multi procs
this my supervisord config file
```
[program:asteriskEvents]
process_name=%(program_name)s_%(process_num)02d
command=php /var/www/laravel/artisan asterisk:events
autostart=true
autorestart=true
user=asterisk
numprocs=2
redirect_stderr=true
stdout_logfile=/var/log/supervisor/asterisk_events.log
stopwaitsecs=3600
```
this work as lister for Asterisk server events that come with AMI
I need a way to make this working with aws function exactly like how you do this with Queues
Please advice if this possible

**Comment by mnapoli** [comment]
Hi,

To run Laravel Queues on AWS Lambda have a look at the Bref-Laravel Bridge: https://github.com/brefphp/laravel-bridge#laravel-queues-with-sqs

**Reply by marayshi** [reply]
Hi @mnapoli 
Yes i know how to run queues with brefphp
I'm asking for different thing

I mean how to run a-listener for any things on aws with bref
I tries to make it with consolo command to be run in schedule every 5 minutes and its timeout 5 min
and this working but the aws invoice going crazy up because this is function never stop 
not sure if there away to do this without run the function 24/7

**Reply by mnapoli** [reply]
OK, but AWS Lambda is **event-driven**.

It's whole execution model and billing model works around the fact that **code must be triggered by an event**.

You are not supposed to run a daemon inside Lambda. Or said another way: you shouldn't poll and wait for anything.

Could you detail why you absolutely want to do it that way (instead of using the SQS trigger)? What are you trying to achieve? Maybe we can find a different approach together.

**Reply by marayshi** [reply]
I'm trying to connect to Asterisk server and listening for Asterisk calls that come by sockets
I doing this by create my sockets code as laravel console command 
and run supervisor service that executing this command
it should keep listening events that come any time from asterisk 
and this working good within Ec2 
I think no way to do this as lambda function

But what make me asking this question, that we do the laravel queues in same way with supervisor 
and now it working with lambda in perfect way, but things different with queues that we know when to dispatch the queue, but with my case i dont know the time when events come from asterisk, so i have to keep listening

i hope this help to understand

**Reply by matthewsuan** [reply]
AFAIK, there is no serverless socket client offering from AWS (I wish there was!). 

I think another option is to use Fargate as listener then push events to SQS or SNS or Kinesis. Then have lambda process each event. Or process each event inside Fargate directly.

**Reply by marayshi** [reply]
Thank's @matthewsuan 
I will look into Fargate

**Reply by mnapoli** [reply]
Yep, that doesn't sound like a good fit on Lambda to be honest.

Fargate could be an option but I'm not sure you would gain a lot compared to EC2, it might be more expensive if you run all the time.

But nothing prevents you from running the socket watcher on EC2 and push jobs in SQS, then Lambda processes SQS. That way you can have a really small script on EC2 without Laravel, and keep only Laravel in Lambda. But it's up to you.

**Reply by marayshi** [reply]
Yes you right, this not fit to Lambda
I already doing this by EC2 and it working good, I just dreamed to put all my code parts in one place "Lambda" haha
Thank's @mnapoli 👍
